### PR TITLE
chore: add Cypress e2e tests

### PR DESCRIPTION
**Pull request by astone123**
This PR configures Cypress e2e testing and adds one simple test. It also configures CI to cache dependencies, build artifacts, and Cypress binaries, and to run Cypress e2e tests. 

Note that because I'm running a fork, the Cypress record key secret isn't available to me. It will only run and record to Cypress Cloud in the main repo. Trying to figure out how to share these secrets to forks safely.

Planning to add more e2e tests in a separate PR.

### Testing
```
bun install
```
```
bun run test:e2e
```

### CODE REVIEW

File: cypress.config.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import {defineConfig} from 'cypress';
+
+export default defineConfig({
+  projectId: 'n3os69',
```
Comment (wweitzel) [inline cypress.config.ts:4]: Where is this value coming from?

File: cypress/support/commands.ts
Comment (wweitzel) [inline cypress/support/commands.ts:1]: Do we need this file? The whole thing is commented out.

File: cypress/support/e2e.ts
Comment (wweitzel) [inline cypress/support/e2e.ts:1]: I don't think we need this either. So I guess this whole /support folder can go?

File: cypress.config.ts (hunk 1 above)
Comment (astone123) [reply cypress.config.ts:4]: This comes from our [Cypress Cloud project](https://cloud.cypress.io/projects/n3os69/settings)

File: cypress/support/commands.ts
Comment (astone123) [reply cypress/support/commands.ts:1]: We don't need it right now, I figured I'd keep it as a reference for how to add custom commands to Cypress as we add more tests

File: cypress/support/e2e.ts
Comment (astone123) [reply cypress/support/e2e.ts:1]: Yeah could delete this for now. Might be a good reference though

File: cypress/support/commands.ts
Comment (wweitzel) [reply cypress/support/commands.ts:1]: I would prefer removing these. Don't want to have files and folders that are not doing anything and cypress has docs for this stuff https://docs.cypress.io/api/cypress-api/custom-commands

File: cypress/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Cypress Tests
```
Comment (wweitzel) [inline cypress/README.md:1]: Can we please add this information to the README we have in the root directory? And then remove this file?

And for folder structure, I'd like to keep it simple and have one folder called `e2e` in the root directory and that folder just has one file in it `goals.cy.ts`.

File: .github/workflows/build-and-test.yml [hunk 3]
```diff
@@ -5,19 +5,94 @@ on:
...
+      - name: Restore cached dependencies
+        id: dependency-restore
+        uses: actions/cache/restore@v3
+        with:
+          path: node_modules/
+          key: ${{ runner.os }}-node-modules-${{ hashFiles('bun.lockb') }}
       - name: Install dependencies
-        run: bun i
+        run: bun i --frozen-lockfile
+      - name: Save dependency cache
```
Comment (wweitzel) [inline .github/workflows/build-and-test.yml:24]: The downsides of bun have begun... can't use the cypress github action. If stuff like this keeps coming up we are going to have to switch back to node/npm.

File: cypress/support/e2e.ts
Comment (wweitzel) [reply cypress/support/e2e.ts:1]: Definitely prefer removing to keep the code base slim

File: .github/workflows/build-and-test.yml (hunk 3 above)
Comment (astone123) [reply .github/workflows/build-and-test.yml:24]: yeah.... 😬

**Comment by astone123** [comment]
Alright things are looking better now @wweitzel

File: .github/workflows/build-and-test.yml (hunk 3 above)
Comment (wweitzel) [reply .github/workflows/build-and-test.yml:24]: 🙃

**Review by wweitzel — APPROVED** [review]
Nice to have the foundation for e2e tests now! And thanks making the changes

**Pull request merged**